x/farming/types/msgs.go

```diff
@@ -56,13 +56,13 @@ func (msg MsgCreateFixedAmountPlan) ValidateBasic() error {
 		return sdkerrors.Wrapf(ErrInvalidPlanEndTime, "end time %s must be greater than start time %s", msg.EndTime.Format(time.RFC3339), msg.StartTime.Format(time.RFC3339))
 	}
 	if msg.StakingCoinWeights.Empty() {
-		return ErrEmptyStakingCoinWeights
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "staking coin weights must not be empty")
 	}
 	if err := msg.StakingCoinWeights.Validate(); err != nil {
```

Comment:
Can we add validation for checking the sum of total weights == 1?

Reply:
I think it should also be included in `MsgCreateRatioPlan`, `MsgCreateFixedAmountPlan`, `AddRequestProposals`, , `UpdateRequestProposals` and `func (plan BasePlan) Validate()`

Reply:
It looks like validation logic is not used for `Stake`, `Unstake`, `Harvest` as well. I'll add those  too.